File: 30-Days-of-Code/Day-2-Operators/Javascript/solution.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

function main() {
    
    var mealCost;
    var tipPercent;
    var taxPercent;

    mealCost = parseFloat(readLine());
    tipPercent = parseInt(readLine());
    taxPercent = parseInt(readLine());

    var tip = mealCost * (tipPercent / 100.0);
    var tax = mealCost * (taxPercent / 100.0);
    var totalCost = mealCost + tip + tax;
    
    var roundedTotalCost = Math.round(totalCost);
    
    process.stdout.write("The total meal cost is " + roundedTotalCost + " dollars.\n")
    
}

File: 30-Days-of-Code/Day-2-Operators/Scala/solution.scala
import math.round

object Solution {
    def main(args: Array[String]) {
        val mealCost = scala.io.StdIn.readDouble()
        val tipPresent = scala.io.StdIn.readInt()
        val taxPresent = scala.io.StdIn.readInt()
            
        val tip = mealCost * (tipPresent / 100.0)
        val tax = mealCost * (taxPresent / 100.0)
        val totalCost = mealCost + tip + tax
            
        val roundedTotalCost = round(totalCost)

        printf("The total meal cost is %d dollars.", roundedTotalCost)
    }
}
